### DISCUSSION TITLE
딥 링크 처리 (카카오톡 링크 공유)

### DISCUSSION THREAD

**Discussion by junyng** [Team-Planz/Planz-iOS#36] [General]
유저가 카카오 톡 링크를 통해 앱을 열었을 때, 다음의 화면이 나타나게 됩니다.

1. 로그인을 하지 않았을 때, 로그인 화면 보여주기
2. 약속 응답화면 보여주기

위의 딥링크 처리는 아래의 과정을 거치게 될 것 같은데요
1. AppDelegate `application(_ UIApplication: , open: URL, options: [UIApplication.OpenURLOptionsKey: Any] = [:])` 에서 이벤트가 발생
2. `AppStore<AppState, AppAction>`를 통해 액션을 받아옴
3. `AppState.ChildState.presentationState`를 변경해줌으로써 화면 전환

이 과정에서 문제될 만한 과정이 있을까요?

**Comment by GangWoon** [comment]
AppDelegate 사용을 피할 수 있을 거 같습니다.
``` swift
        WindowGroup {
            ContentView()
                .onOpenURL { store.send(.openURL(url: $0)  }
        }
```

WindowGroup에서 .sheet 뷰 모디파이러를 바인딩했을 때, 저희가 원하는대로 상태값을 바꼈을 때, sheet가 나오는 지 확인이 필요해보입니다.
